client/product-details/index.js

```diff
@@ -7,51 +7,95 @@
 import { initializeBnplSiteMessaging } from './bnpl-site-messaging';
 
 jQuery( function ( $ ) {
+	/**
+	 * Check for the existence of the `wcpayStripeSiteMessaging` variable on the window object.
+	 * This variable holds the configuration for Stripe site messaging and contains the following keys:
+	 *  - productId: The ID of the product.
+	 *  - productVariations: Variations of the product.
+	 *  - country: The country of the customer. Defaults to the store's country.
+	 *  - publishableKey: The key used for Stripe's API calls.
+	 *  - paymentMethods: Enabled BNPL payment methods.
+	 *
+	 * If this variable is not set, the script will exit early to prevent further execution.
+	 */
+	if ( ! window.wcpayStripeSiteMessaging ) {
+		return;
+	}
+
+	const { productVariations, productId } = window.wcpayStripeSiteMessaging;
 	const bnplPaymentMessageElement = initializeBnplSiteMessaging();
-	const { productVariations } = window.wcpayStripeSiteMessaging;
-	let { productId } = window.wcpayStripeSiteMessaging;
+
+	// Utility function to safely parse integers and handle NaN
+	const safeParseInt = ( value ) => {
+		const result = parseInt( value, 10 );
+		return isNaN( result ) ? 0 : result;
+	};
+
+	const updateBnplPaymentMessage = ( amount, currency ) => {
+		if ( amount > 0 && currency ) {
+			bnplPaymentMessageElement.update( { amount, currency } );
+		}
+	};
+
+	const quantityInput = $( '.quantity input[type=number]' );
 
 	const resetBnplPaymentMessage = () => {
-		const quantity = $( '.quantity input[type=number]' ).val();
-		productId = 'base_product';
-		bnplPaymentMessageElement.update( {
-			amount:
-				parseInt( productVariations.base_product.amount, 10 ) *
-				quantity,
-			currency: productVariations.base_product.currency,
-		} );
+		if ( ! quantityInput.length || ! productVariations.base_product )
```

Comment:
We are missing `{}` here.

Reply:
Done here https://github.com/Automattic/woocommerce-payments/pull/7153/commits/15d2d7d9d496dd3f5f60de7c92f249cf56e6c348 thanks!